superset/assets/javascripts/explore/components/AdhocMetricEditPopover.jsx

```diff
@@ -40,6 +59,22 @@ export default class AdhocMetricEditPopover extends React.Component {
       clearable: true,
       selectWrap: VirtualizedSelect,
     };
+    if (langTools) {
+      const words = sqlWords.concat(this.props.columns.map(column => (
+        { name: column.column_name, value: column.column_name, score: 50, meta: 'column' }
+      )));
+      const completer = {
+        getCompletions: (aceEditor, session, pos, prefix, callback) => {
+          callback(null, words);
+        },
+      };
+      langTools.setCompleters([completer]);
+    }
+    document.addEventListener('mouseup', this.onMouseUp);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mouseup', this.onMouseUp);
```

Comment:
unlikely sitch, but could also call `document.removeEventListener('mousemove', this.onMouseMove);` to be safe

Reply:
oooh good call